### PR TITLE
Fixes add-model config for CAAS

### PR DESCRIPTION
**Pull request by tlmiller**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

CAAS models would not pass on config changes when adding a new model. This fix combines both IAAS and CAAS model config creation into one logic path now.

## QA steps

See bug https://bugs.launchpad.net/juju/+bug/1855503

## Documentation changes

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1855503

### CODE REVIEW

File: cloud/clouds.go [hunk 1]
```diff
@@ -269,7 +269,11 @@ var caasCloudTypes = map[string]bool{
 
 // CloudIsCAAS checks if cloud is a CAAS cloud.
 func CloudIsCAAS(cloud Cloud) bool {
-	return caasCloudTypes[cloud.Type]
+	return CloudTypeIsCAAS(cloud.Type)
+}
+
+func CloudTypeIsCAAS(cloudType string) bool {
```
Comment (hpidcock) [inline cloud/clouds.go]: Exported functions need a comment.

**Comment by hpidcock** [comment]
[juju add-model config](https://trello.com/c/4GLwhZab/998-juju-add-model-config)

**Comment by hpidcock** [comment]
!!build!!

**Comment by tlmiller** [comment]
$$build$$

**Comment by tlmiller** [comment]
!!build!!

**Comment by hpidcock** [comment]
$$merge$$

**Comment by tlmiller** [comment]
$$merge$$

**Comment by tlmiller** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**